ccat.py

```diff
@@ -354,6 +402,234 @@ def append_configuration(self, answers):
+
+    def get_menu(self):
+        return [
+            Separator('= GCP ='),
+            ENUMERATE_GCR,
+            LIST_GCR_REPOS,
+            PULL_GCR_REPOS,
+            PUSH_GCR_REPOS,
+            SWAP_SERVICE_ACCOUNT
+        ]
+
+    def swap_service_account(self):
+        if self.configuration.get('service_account_json_file_path'):
+            print('Current Service Account: {}'.format(self.configuration.get('service_account_json_file_path')))
+        self.set_configuration()
+
+    def ask_configuration(self):
+        if not self.configuration.get('service_account_json_file_path') and not self.configuration.get('access_token'):
+            print('Did not find GCP configuration!')
+
+        questions = [
+            {
+                'type': 'list',
+                'name': 'creds_choice',
+                'message': 'What GCP credential would you like to setup?',
+                'choices': ['Service Account', 'Access Token']
+            }
+        ]
+
+        answers = prompt(questions)
+
+        if answers['creds_choice'] == 'Service Account':
+            # service account
+            questions = [
+                {
+                    'type': 'input',
+                    'name': 'service_account_json_file_path',
+                    'message': 'Enter GCP Service Account json file path',
+                    'validate': lambda profile: len(profile) != 0 or 'GCP Service Account can not be empty!'
+                }
+            ]
+
+            while True:
+                answers = prompt(questions)
+                path = pathlib.Path(answers.get('service_account_json_file_path'))
+                if path.exists():
+                    break
+                else:
+                    print("Service account json file does not exist")
+        else:
+            # access token
+            questions = [
+                {
+                    'type': 'input',
+                    'name': 'access_token',
+                    'message': 'Enter GCP Access Token',
+                    'validate': lambda profile: len(profile) != 0 or 'GCP Access Token can not be empty!'
+                }
+            ]
+
+            answers = prompt(questions)
+
+        return answers
+
+
+    def is_configured(self):
+        return self.configuration.get('service_account_json_file_path') is not None or self.configuration.get('access_token') is not None
+
+    def set_configuration(self):
+        answers = self.ask_configuration()
+
+        print('Configuring GCP...')    
+    
+        self.configuration.update({
+            'service_account_json_file_path': answers.get('service_account_json_file_path'),
+            'access_token': answers.get('access_token')
+        })
+
+        print('Successfully configured GCP\n')
+
+    def ask_gcr_enum_repos(self):
+        if self.is_configured() is False:
+            self.set_configuration()
+        
+        gcp_registries = ['gcr.io','us.gcr.io','eu.gcr.io','asia.gcr.io']
```

Comment:
Like you said, is there a way to get this list programmatically?

Reply:
I believe, there must be a way to get this programmatically. I will do that when I am possible then I will create another PR for that.